dataplane/slow_worker.h

```diff
@@ -1,22 +1,147 @@
-	void PreparePacket(rte_mbuf* pkt);
+	struct Config
+	{
+		uint32_t SWICMPOutRateLimit = 0;
+		bool use_kernel_interface;
+	};
+
+private:
+	bool m_stop = false;
+	std::vector<tPortId> m_ports_serviced;
+	std::vector<cWorker*> m_workers_serviced;
+	std::vector<dpdk::RingConn<rte_mbuf*>> from_gcs_;
+	cWorker* m_slow_worker;
+	std::queue<
+	        std::tuple<
+	                rte_mbuf*,
+	                common::globalBase::tFlow>>
+	        m_slow_worker_mbufs;
+	rte_mempool* m_mempool; // from cControlPlane
+	common::slowworker::stats_t m_stats;
+	fragmentation::Fragmentation m_fragmentation;
+	dregress_t m_dregress;
+	uint32_t m_icmp_out_remainder;
+	Config m_config;
+	dataplane::KernelInterfaceWorker m_kni_worker;
+
+public:
+	SlowWorker(cWorker* worker,
+	           const std::vector<tPortId>& ports_to_service,
+	           const std::vector<cWorker*>& workers_to_service,
+	           std::vector<dpdk::RingConn<rte_mbuf*>>&& from_gcs,
+	           KernelInterfaceWorker&& kni,
+	           rte_mempool* mempool,
+	           bool use_kni,
+	           uint32_t sw_icmp_out_rate_limit);
+	SlowWorker(const SlowWorker&) = delete;
+	SlowWorker(SlowWorker&& other);
+
+	cWorker* GetWorker() { return m_slow_worker; }
+	const dataplane::KernelInterfaceWorker& KniWorker() const { return m_kni_worker; }
+	const dataplane::base::generation& current_base() { return m_slow_worker->current_base(); }
+	const fragmentation::Fragmentation& Fragmentation() const { return m_fragmentation; }
+	dregress_t& Dregress() { return m_dregress; }
+	const dregress_t& Dregress() const { return m_dregress; }
+	void freeWorkerPacket(rte_ring* ring_to_free_mbuf, rte_mbuf* mbuf);
+	rte_mbuf* convertMempool(rte_ring* ring_to_free_mbuf, rte_mbuf* old_mbuf);
+	rte_mempool* Mempool() { return m_mempool; }
+	void PreparePacket(rte_mbuf* mbuf) { m_slow_worker->preparePacket(mbuf); }
+	void SendToSlowWorker(rte_mbuf* mbuf, const common::globalBase::tFlow& flow);
+	void ResetIcmpOutRemainder(uint32_t limit) { m_icmp_out_remainder = limit; }
+
+	const common::slowworker::stats_t& Stats() const { return m_stats; }
+	unsigned ring_handle(rte_ring* ring_to_free_mbuf, rte_ring* ring);
+	void handlePacket_icmp_translate_v6_to_v4(rte_mbuf* mbuf);
+	void handlePacket_icmp_translate_v4_to_v6(rte_mbuf* mbuf);
+	void handlePacket_fragment(rte_mbuf* mbuf);
+	void handlePacket_dregress(rte_mbuf* mbuf);
+	void handlePacket_farm(rte_mbuf* mbuf);
+	void handlePacket_repeat(rte_mbuf* mbuf);
+	void handlePacket_fw_state_sync(rte_mbuf* mbuf);
+	bool handlePacket_fw_state_sync_ingress(rte_mbuf* mbuf);
+	void handlePacket_balancer_icmp_forward(rte_mbuf* mbuf);
+	void handlePacketFromForwardingPlane(rte_mbuf* mbuf);
+	void HandleWorkerRings();
+
+	// \brief dequeue packets from worker_gc's ring to slowworker
+	void DequeueGC();
+
+	void WaitInit();
+	void Start()
+	{
+		WaitInit();
+		while (!m_stop)
+		{
+			Iteration();
+		}
+	}
```

Comment:
I couldn't find any code that changes `m_stop`. So this seems like an infinite loop for me

Reply:
Start() is also unused, removed